scalding-core/src/main/scala/com/twitter/scalding/macros/impl/FieldsProviderImpl.scala

```diff
@@ -80,53 +80,81 @@ object FieldsProviderImpl {
-        This will mean the macro is operating on a non-resolved type.""")
-
-    def matchField(fieldType: Type, outerName: Option[String], fieldName: String, isOption: Boolean): List[(Tree, String)] = {
-      val returningType = if (isOption) q"""classOf[java.lang.Object]""" else q"""classOf[$fieldType]"""
-      val simpleRet = outerName match {
-        case Some(outer) => List((returningType, s"$outer$fieldName"))
-        case None => List((returningType, s"$fieldName"))
-      }
-      fieldType match {
-        case tpe if tpe =:= typeOf[String] => simpleRet
-        case tpe if tpe =:= typeOf[Boolean] => simpleRet
-        case tpe if tpe =:= typeOf[Short] => simpleRet
-        case tpe if tpe =:= typeOf[Int] => simpleRet
-        case tpe if tpe =:= typeOf[Long] => simpleRet
-        case tpe if tpe =:= typeOf[Float] => simpleRet
-        case tpe if tpe =:= typeOf[Double] => simpleRet
-        case tpe if tpe.erasure =:= typeOf[Option[Any]] && isOption == true => c.abort(c.enclosingPosition, s"Case class ${T} has nested options, not supported currently.")
-        case tpe if tpe.erasure =:= typeOf[Option[Any]] =>
-          val innerType = tpe.asInstanceOf[TypeRefApi].args.head
-          matchField(innerType, outerName, fieldName, true)
-        case tpe if IsCaseClassImpl.isCaseClassType(c)(tpe) =>
-          val prefix = outerName.map(pre => s"$pre$fieldName.")
-          expandMethod(tpe, prefix, isOption)
-        case tpe if allowUnknownTypes => simpleRet
-        case _ => c.abort(c.enclosingPosition, s"Case class ${T} is not pure primitives or nested case classes")
+    def maybeHandlePrimitive(outerTpe: Type): Option[List[(Tree, String)]] = {
+
+      def innerLoop(tpe: Type, inOption: Boolean): Option[List[(Tree, String)]] = {
+        val returningType = if (inOption) q"""classOf[java.lang.Object]""" else q"""classOf[$outerTpe]"""
+        val simpleRet = Some(List((returningType, s"element")))
+
+        tpe match {
+          case tpe if tpe =:= typeOf[String] => simpleRet
+          case tpe if tpe =:= typeOf[Boolean] => simpleRet
+          case tpe if tpe =:= typeOf[Short] => simpleRet
+          case tpe if tpe =:= typeOf[Int] => simpleRet
+          case tpe if tpe =:= typeOf[Long] => simpleRet
+          case tpe if tpe =:= typeOf[Float] => simpleRet
+          case tpe if tpe =:= typeOf[Double] => simpleRet
+          case tpe if tpe.erasure =:= typeOf[Option[Any]] && inOption == true => c.abort(c.enclosingPosition, s"Case class ${T} has nested options, not supported currently.")
+          case tpe if tpe.erasure =:= typeOf[Option[Any]] =>
+            val innerType = tpe.asInstanceOf[TypeRefApi].args.head
+            innerLoop(innerType, true)
+
+          case tpe => None
+        }
       }
+      innerLoop(outerTpe, false)
     }
+    val expanded: List[(Tree, String)] = maybeHandlePrimitive(T.tpe).getOrElse {
```

Comment:
see the link to the old TypedTsv code. It used numbers not names and I
think cascading treats numbers differently.

So, if we use 0, we are doing the same thing we were doing before, so
hopefully the tests would pass.

Also, we may need to do the same thing for tuples since there could be
similar failures at runtime for Fields-API users using TypedTsv[(Int, Int)].

On Tuesday, July 14, 2015, Sidharth Kapur <notifications@github.com> wrote:

> In
> scalding-core/src/main/scala/com/twitter/scalding/macros/impl/FieldsProviderImpl.scala
> <https://github.com/twitter/scalding/pull/1356#discussion_r34633126>:
>
> >      }
> > +    val expanded: List[(Tree, String)] = maybeHandlePrimitive(T.tpe).getOrElse {
>
> Why 0 instead of "element"?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/twitter/scalding/pull/1356/files#r34633126>.
>


-- 
Oscar Boykin :: @posco :: http://twitter.com/posco